src/auth/basic/SASL/required.m4

```diff
@@ -6,4 +6,11 @@
 ##
 
 SQUID_CHECK_SASL
-AS_IF([test "x$squid_cv_check_sasl" = "xyes"],[BUILD_HELPER="SASL"])
+AS_IF([test "x$squid_cv_check_sasl" = "xyes"],[
+    BUILD_HELPER="SASL"
+])
+AS_IF([test "x$squid_host_os.$squid_cv_check_sasl" = "xdarwin.yes"],[
+  AS_IF([test "x$squid_cv_cc_arg_wno_deprecated_declarations" != "xyes"],[
```

Comment:
> IMO preventing duplicate flags is a larger issue 

I agree that it is a separate issue of sufficient "size" to warrant a dedicated PR.

> that affects a lot of things out of this PR scope

Preventing duplicate flags may affect a lot of future changes. However, it does not affect anything in current official code because we never explicitly call SQUID_CC_ADD_CXXFLAG_WARNING_IF_SUPPORTED() twice with the same argument AFAICT. This is a (suspected) new problem, (suspected to be) introduced by this PR changes. Moreover, it is possible to refactor this PR to avoid (suspicions about introducing) this problem in the first place.

> I believe the refactor and checking is best done elsewhere, so am against complicating this PR with the request to refactor.

I support (and even encourage) enhancing SQUID_CC_ADD_CXXFLAG_WARNING_IF_SUPPORTED (or whatever) to avoid flags duplication in a dedicated PR. It is possible to test those enhancements properly without this PR changes (e.g., by temporary introducing duplicated calls), and I hope that Francesco will do that.

I do not support merging this PR until this issue is resolved (one way or another), especially if this PR may result in duplicated flags. Fortunately, there is no rush with landing this PR AFAICT. We can wait for the prerequisite do-not-duplicate-flags PR to land first. Thus, there is currently no conflict in the associated change requests.

Reply:
PR branch commit 0d9e75fc8129d29f6e534c5801a0de2060910f67 removed flags/warning manipulation, obsoleting this change request.

Reply:
That's correct. In the end @yadij 's rework of lib detection simplified things tremendously and
uncovered the actual problem - an overcomplicated SASL detection.